tests/src/Unit/SocialApiTest.php

```diff
@@ -0,0 +1,560 @@
+  }
+
+  /**
+   * Test for class SocialApiImplementerInstaller
+   */
+
+  public function testSocialApiImplementerInstaller () {
+    // assertion to check if the file exists
+    // $this->assertFileExists('../drupal8/modules/social_api/src/Utility/SocialApiImplementerInstaller.php');
+
+    $collection = new SocialApiImplementerInstaller();
+    // mock object to our interface
+    $mock = $this->getMock('Drupal\social_api\Utility\SocialApiImplementerInstaller');
+    // check if the mock object belongs to our interface
+    $this->assertTrue($mock instanceof SocialApiImplementerInstaller);
+  }
+
+  /**
+   * Test for interface UserManagerInterface
+   */
+
+   public function testUserManagerInterface () {
+     // assertion to check if the file exists
+     // $this->assertFileExists('../drupal8/modules/social_api/src/User/UserManagerInterface.php');
+
+     $collection = $this->createMock(UserManagerInterface::class);
+     $this->assertTrue($collection instanceof UserManagerInterface);
+     $this->assertTrue(
+       method_exists($collection, 'getPluginId'),
+       'UserManagerInterface does not have getPluginId function/method'
+     );
+     $this->assertTrue(
+       method_exists($collection, 'setPluginId'),
+       'UserManagerInterface does not have fsetPluginId function/method'
+     );
+     $this->assertTrue(
+       method_exists($collection, 'getDrupalUserId'),
+       'UserManagerInterface does not have getDrupalUserId function/method'
+     );
+
+   }
+
+   /**
+    * Test for class UserManager
+    */
+
+   public function testUserManager () {
+     // assertion to check if the file exists
+     // $this->assertFileExists('../drupal8/modules/social_api/src/User/UserManager.php');
+
+     // creating a mock object and passing constructor paremeters
+     $this->entity_type_manager = $this->createMock(EntityTypeManagerInterface::class);
+     $this->messenger = $this->createMock(MessengerInterface::class);
+     $this->logger_factory = $this->createMock(LoggerChannelFactoryInterface::class);
+
+     $collection = $this->getMockBuilder('Drupal\social_api\User\UserManager')
+                        ->setConstructorArgs(array($this->entityType, $this->entity_type_manager, $this->messenger, $this->logger_factory))
+                        ->getMockForAbstractClass();
+     $this->assertTrue($collection instanceof UserManager);
+     // checking for correct setPluginId and getPluginId method
+     $collection->setPluginId('1234');
+     // var_dump($collection->getSessionPrefix());
+     $this->assertEquals('1234', $collection->getPluginId());
+   }
+
+   /**
+    * Test for class UserAuthenticator
+    */
+
+   public function testUserAuthenticator () {
+     // assertion to check if the file exists
+     // $this->assertFileExists('../drupal8/modules/social_api/src/User/UserAuthenticator.php');
+
+     // creating a mock object and passing constructor paremeters
+     $this->sessionKeys = array(1,2,3);
```

Comment:
Sessions key are not usually numbers, they are strings. Rule of thumb, make your tests as approximate as possible to the real scenario.